connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.connect.connector.Connector;
+import org.apache.kafka.connect.connector.ConnectorContext;
+import org.apache.kafka.connect.sink.SinkConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class WorkerConnector {
+    private static final Logger log = LoggerFactory.getLogger(WorkerConnector.class);
+
+    private enum State {
+        STOPPED, STARTED, FAILED
+    }
+
+    private final String connName;
+    private final ConnectorStatus.Listener statusListener;
+    private final ConnectorContext ctx;
+    private final Connector connector;
+
+    private Map<String, String> config;
+    private State state;
+
+    public WorkerConnector(String connName,
+                           Connector connector,
+                           ConnectorContext ctx,
+                           ConnectorStatus.Listener statusListener) {
+        this.connName = connName;
+        this.ctx = ctx;
+        this.connector = connector;
+        this.statusListener = statusListener;
+        this.state = null;
+    }
+
+    public synchronized void initialize(Map<String, String> config) {
+        log.debug("Initializing connector {} with config {}", connName, config);
+
+        this.config = config;
+
+        connector.initialize(new ConnectorContext() {
+            @Override
+            public void requestTaskReconfiguration() {
+                ctx.requestTaskReconfiguration();
+            }
+
+            @Override
+            public void raiseError(Exception e) {
+                WorkerConnector.this.state = State.FAILED;
+                ctx.raiseError(e);
+            }
+        });
+    }
+
+    private void start() {
+        try {
+            if (state == State.STARTED)
```

Comment:
Is it worth handling all of these cases with enums so we have a way to make sure any new states are properly handled when they are added? Not sure we'll ever add any new states, but if we do it might be a good safety net.